requests/utils.py

```diff
@@ -447,6 +446,29 @@ def get_encodings_from_content(content):
             xml_re.findall(content))
 
 
+def _parse_content_type_header(header):
+    """Returns content type and parameters from given header
+
+    :param header: string
+    :return: tuple containing content type and dictionary of
+         parameters
+    """
+
+    tokens = header.split(';')
+    content_type, params = tokens[0].strip(), tokens[1:]
+    params_dict = {} 
+
+    for param in params:
+        if param and not param.isspace():
+            param = param.strip()
+            key, value = param, True
+            if '=' in param:
```

Comment:
Might be able to get rid of the list comprehension here by using `param.index('=')` and checking for a non-negative? Definitely would be faster. Something to the tune of:

```python
key, value = param, True
index = param.index('=')
if index != -1:
    key = param[:index].strip('\'" ')
    value = param[index+1:].strip('\'" ')
```